src/components/interactive-map.js

```diff
@@ -121,102 +168,125 @@ export default class InteractiveMap extends PureComponent {
   }
 
   _handleEvent(event) {
-    const {mapControls} = this.props;
-    // MapControls only extracts the states that it recognizes.
-    // This allows custom map controls to add new states and callbacks.
-    mapControls.setState(Object.assign({}, this.props, {
-      mapState: new MapState(this.props)
-    }));
-
-    return mapControls.handle(event);
+    const controlOptions = Object.assign({
+      onChangeState: this._onInteractiveStateChange
+    }, this.props);
+    return this.props.mapControls.handle(event, controlOptions);
   }
 
-  // TODO - Remove once Viewport alternative is good enough
-  _getMap() {
-    return this._map._getMap();
+  getMap() {
+    return this._map.getMap();
   }
 
-  // Calculate a cursor style to show that we are in "dragging state"
-  _getCursor() {
-    const isInteractive =
-      this.props.onChangeViewport ||
-      this.props.onClickFeature ||
-      this.props.onHoverFeatures;
-
-    if (!isInteractive) {
-      return 'inherit';
-    }
-    if (this.props.isDragging) {
-      return config.CURSOR.GRABBING;
-    }
-    if (this.props.isHovering) {
-      return config.CURSOR.POINTER;
-    }
-    return config.CURSOR.GRAB;
+  queryRenderedFeatures(geometry, options) {
+    return this._map.queryRenderedFeatures(geometry, options);
   }
 
-  // Checks a displayConstraints object to see if the map should be displayed
-  checkDisplayConstraints(props) {
+  // Checks a visibilityConstraints object to see if the map should be displayed
+  checkVisibilityConstraints(props) {
     const capitalize = s => s[0].toUpperCase() + s.slice(1);
 
-    const {displayConstraints} = props;
+    const {visibilityConstraints} = props;
     for (const propName in props) {
       const capitalizedPropName = capitalize(propName);
       const minPropName = `min${capitalizedPropName}`;
       const maxPropName = `max${capitalizedPropName}`;
 
-      if (minPropName in displayConstraints && props[propName] < displayConstraints[minPropName]) {
+      if (minPropName in visibilityConstraints &&
+        props[propName] < visibilityConstraints[minPropName]) {
         return false;
       }
-      if (maxPropName in displayConstraints && props[propName] > displayConstraints[maxPropName]) {
+      if (maxPropName in visibilityConstraints &&
+        props[propName] > visibilityConstraints[maxPropName]) {
         return false;
       }
     }
     return true;
   }
 
+  _getFeatures({pos, radius}) {
```

Comment:
As stated would like another round of discussion before we move model event handling out of the model class. The remaining changes are fine.